src/main/kotlin/account/CaixaEletronico.kt

```diff
@@ -0,0 +1,62 @@
+package account
+
+class CaixaEletronico(
+    private val account: Account
+){
+
+    fun abreGaveta(gaveta: Gaveta): String {
+
+        return when(gaveta){
+            Gaveta.DEPOSITO -> "abrindo gaveta para receber valor para deposito"
+            Gaveta.SAQUE -> "abrindo gaveta para emitir o valor solicitado para saque"
+            Gaveta.DEVOLUCAO -> "Abrindo gaveta para devolver dinheiro"
+        }
+    }
+
+    fun recebeDinheiro(valor: Double): String {
```

Comment:
Vou tangibilizar um pouco melhor o problema porque acho que não deixei claro como funciona a operação física do caixa eletrônico que propus.

Depósito:

- No início, o cliente digita o valor que quer depositar no caixa eletrônico. As devidas validações de valores (se está no range válido) são feitas e, em caso de erro, o devido tratamento é feito;

- Em seguida, a gaveta abre para receber o valor que o cliente digitou;

- Cliente insere o dinheiro.

- É realizada a contabilização do dinheiro que o cliente inseriu. Se valor contado pela máquina for diferente do inserido pelo cliente, devo tratar esse cenário (**como?**);

- Caso contrário, dinheiro é somado na conta do cliente

- Caso de uso é encerrado;

Algumas dicas:

- A classe `Account` só deveria ser responsável por somar ou subtrair valores na conta do cliente (i.e.: dado um `valor`, faço as verificações e, se elas passarem, guardo na memória o valor somado ou subtraído de dinheiro). Ao que entendi, foi isso mesmo que você fez, mas os nomes dos métodos `deposita` e `saque` deixaram isso um pouco confuso, pq deu a entender que toda a operação de depósito (abertura da gaveta, comparação de valores, contabilização do resultado) deveria estar lá. Acha que faz sentido mudar os nomes?

- Note que estamos lidando com 2 valores: **um informado pelo cliente** e outro **contado pela máquina**. O primeiro você pode tratar como já fez (como uma variável externa `valor`) e o segundo você pode assumir que a própria máquina fornece esse método;

- Sugestão: isole a classe `Gaveta` e assuma que ela tem as seguintes responsabilidades:
  - Abrir();
  - Fechar();
  - ContarDinheiroInserido();
  - Guardar a informação de quanto dinheiro inserido tem nela;

Reply:
Deixo um cenários de teste como exemplo, pra tangibilizar o que espero do caixa:

`dado que o cliente informou um valor **válido** e inseriu menos (mais) dinheiro que o informado, eu quero que...` 1) `o caixa informe o cliente que o valor está incorreto`;
2) `o caixa devolva o dinheiro ao cliente`;
3) `não entre dinheiro na conta do cliente`;

(Conversamos melhor isso no nosso 1:1 e a gente vê o que você já fez ou tentou fazer aqui)